Signal/src/call/UserInterface/CallUIAdapter.swift

```diff
@@ -94,6 +102,19 @@ class CallUIAdapter {
     }
 
     internal func toggleMute(call: SignalCall, isMuted: Bool) {
+        // With CallKit, muting is handled by a CXAction, so it must go through the adaptee
         adaptee.toggleMute(call: call, isMuted: isMuted)
     }
+
+    internal func toggleSpeakerphone(call: SignalCall, isEnabled: Bool) {
+        // Speakerphone is not handled by CallKit (e.g. there is no CXAction), so we handle it w/o going through the 
+        // adaptee, relying on the AudioService CallObserver to put the system in a state consistent with the call's 
+        // assigned property.
```

Comment:
> Huh, so we can't keep the speakerphone state in sync?

We **aren't**. But I think we could.

A solution might involve using notification center to observe changes in the audio session route. The details of which I haven't solidified, as I think it's lower priority. To be clear this would affect people who toggle speakerphone on callkit, and then switch to the system call screen. (The reverse, where someone toggles speakerphone in app then switches to CallKit works, presumably because the system call screen is doing something outside of CallKit, like observing the AudioSession route.